src/components/swapv2/LimitOrder/OrderBook/OrderItem.tsx

```diff
@@ -48,20 +58,33 @@ export default function OrderItem({
   reverse,
   order,
   style,
+  showAmountOut,
 }: {
   reverse?: boolean
   order: LimitOrderFromTokenPairFormatted
   style: CSSProperties
+  showAmountOut: boolean
 }) {
   const theme = useTheme()
   const upToSmall = useMedia(`(max-width: ${MEDIA_WIDTHS.upToSmall}px)`)
   const { currencyIn, currencyOut } = useLimitState()
+  const { supportedChains } = useChainsConfig()
+
+  const chain = useMemo(
+    () => supportedChains.find(chain => chain.chainId === order.chainId),
+    [order.chainId, supportedChains],
+  )
 
   return (
     <ItemWrapper style={style}>
+      <ChainImage src={chain?.icon} alt="Network" />
       <Rate reverse={reverse}>{order.rate}</Rate>
-      <AmountInfo plus={reverse} amount={order.firstAmount} currency={currencyIn} upToSmall={upToSmall} />
-      <AmountInfo plus={!reverse} amount={order.secondAmount} currency={currencyOut} upToSmall={upToSmall} />
+      {(!upToSmall || !showAmountOut) && (
+        <AmountInfo plus={reverse} amount={order.firstAmount} currency={currencyIn} upToSmall={upToSmall} />
+      )}
+      {(!upToSmall || showAmountOut) && (
+        <AmountInfo plus={!reverse} amount={order.secondAmount} currency={currencyOut} upToSmall={upToSmall} />
+      )}
```

Comment:
is this equivalent with:
```
{!upToSmall && <AmountInfo plus={showAmountOut  ? !reverse : reverse} amount={order.firstAmount} currency={currencyIn} upToSmall={upToSmall} />}

```

Reply:
No, It's not. But to make it easier to understand I will change it into
```
{!upToSmall ? (
    <>
      <AmountInfo plus={reverse} amount={order.firstAmount} currency={currencyIn} upToSmall={upToSmall} />
      <AmountInfo plus={!reverse} amount={order.secondAmount} currency={currencyOut} upToSmall={upToSmall} />
    </>
  ) : (
    <AmountInfo
      plus={showAmountOut ? !reverse : reverse}
      amount={showAmountOut ? order.secondAmount : order.firstAmount}
      currency={showAmountOut ? currencyOut : currencyIn}
      upToSmall={upToSmall}
    />
  )}
```